### PR TITLE
Rewrite mutation code to not need to retry in case of islands

### LINKED ISSUE #1796: Rewrite mutation code to not need to retry in case of islands
As that approach can eventually fail to run, it'd be nice to write a new species mutate method that is built with the island check in mind and it doesn't need to retry the entire mutation attempt if there is a problem.
As the current 10 attempts limit can be reached quite easily, if the player makes very silly cells with a lot of thin cytoplasm strands, a new algorithm that doesn't have this problem should be made.

### PR DESCRIPTION
**Pull request by takko**
**Brief Description of What This PR Does**

added code which moves the islands to the base.

**Related Issues**

closes #1796

**Progress Checklist**

Note: before starting this checklist the PR should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed (this and further items should not be checked by the PR author)
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/general/Mutations.cs
Comment (adQuid) [inline src/general/Mutations.cs:173]: Is this still needed without iterations to wipe away?

File: src/general/Mutations.cs
Comment (adQuid) [inline src/general/Mutations.cs:282]: why does distance == 2 mean early exit? And is that the same as minDistance == 2?

**Review by adQuid — COMMENTED** [review]
Not able to run proper tests right now, and not an expert in hex coordinate math, but I've just got two comments in the code. This isn't code I would expect to fail silently, so I'm happy to approve if people test this enough times and nothing crashes.

File: src/general/Mutations.cs
Comment (takko) [reply src/general/Mutations.cs:173]: This list is provided as parameter and needs to be cleared here, as it contains the copied organelles from the parent.

File: src/general/Mutations.cs
Comment (takko) [reply src/general/Mutations.cs]: Updated the comment to make it more clear.
distance 1 == direct neighbor, distance 2 == min distance for island
distance is always >= minDistance, but changed the comment

File: src/general/Mutations.cs
Comment (adQuid) [reply src/general/Mutations.cs:173]: We provide a list as a param, specifically to clear it out? Sounds like we can just drop the param and declare the variable in here.

Comment (hhyyrylainen) [reply src/general/Mutations.cs:173]: It's easier to pass in the mutated list of organelles. As the mutated species is created by cloning, so it would actually be more work to specifically remove the organelles from it to process here.

**Comment by hhyyrylainen** [comment]
Unless someone else can take a look at this, I'll merge this immediately after 0.5.6.1 to give this time to have more people experience the game with this in it.

File: src/general/Mutations.cs
Comment (hhyyrylainen) [inline src/general/Mutations.cs:260]: Does this work correctly if there are multiple islands on different sides of the microbe? What if those are exactly symmetrical and the average distance is zero?

That's basically my only code concern. This PR basically just needs a lot of testing to make sure it never fails.

**Comment by hhyyrylainen** [comment]
This needs conflict resolving and then testing to make sure this works fine. I'm sure @RacerBG can get some testing done once master is merged into this branch.

**Comment by buckly90** [comment]
I gave it a spin and did not notice any errors in the autoevo notes throughout 25 generations

I was originally aiming for about 30 generations but auto-evo became unbearably slow near the end. Not sure if that is related to this PR or not.

**Comment by hhyyrylainen** [comment]
I guess this PR can slightly increase the auto-evo running time. Or if things get unlucky maybe a lot because 5 mutations are generated for each species so if this doubles the time it takes to generate mutations (which I doubt), it could be impactful. Though I suspect the overall auto-evo running time is still within 5-10% of what it is one the master branch at this point.

Actually as the conflict is not yet fixed, master has not been merged into this, so it might be that my auto-evo running performance improvements aren't included here.

**Comment by RacerBG** [comment]
> I guess this PR can slightly increase the auto-evo running time. Or if things get unlucky maybe a lot because 5 mutations are generated for each species so if this doubles the time it takes to generate mutations (which I doubt), it could be impactful. Though I suspect the overall auto-evo running time is still within 5-10% of what it is one the master branch at this point.
> 
> Actually as the conflict is not yet fixed, master has not been merged into this, so it might be that my auto-evo running performance improvements aren't included here.

I will test this PR after the merge. In terms of lag and perfomance issues I'm probably the most suited one to tell the difference thanks to my hardware.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
Needs to have the merge conflict fixed and playtesting before this can proceed.

**Pull request merged**
